### PR TITLE
Implement Heal Bell, Aromatherapy, Sparkly Swirl and added status effect options to overrides

### PR DESCRIPTION
**Pull request by innerthunder**
This is a cleaner version of my last pull request...

A new attribute `HealPartyStatusAttr` implements the status healing effects of these moves:
- Aromatherapy
- Heal Bell
- Sparkly Swirl

These moves are also implemented/updated with the existing `HealStatusEffectAttr` attribute:
- Purify
- Refresh (should now wake up the user if used via Sleep Talk)

New overrides are also added to `overrides.ts`. `STATUS_OVERRIDE` initalizes player Pokemon with the specified status effect, and `OPP_STATUS_OVERRIDE` does the same for enemy Pokemon.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -35,4 +35,6 @@ src/data/battle-anim-raw-data*.ts
 src/data/battle-anim-data.ts
 src/overrides.ts
 
-coverage
\ No newline at end of file
+coverage
+package-lock.json
+package.json
```
Comment (contra1337) [inline .gitignore:40]: these files are necessary for the project

File: package.json [hunk 2]
```diff
@@ -22,12 +24,13 @@
...
 		"pokenode-ts": "^1.20.0",
 		"typescript": "^5.0.3",
-		"vite": "^4.5.0",
+		"vite": "^4.5.3",
 		"vite-plugin-fs": "^0.4.4",
 		"vitest": "^1.4.0",
 		"vitest-canvas-mock": "^0.3.3"
 	},
 	"dependencies": {
+		"@koa/router": "^12.0.1",
```
Comment (contra1337) [inline package.json:33]: why is this being added? doesn't seem relevant or imported anywhere either

Comment (innerthunder) [reply package.json:33]: Sorry for the confusion, I had an error when I first ran the dev build related to koa/router, and I had to manually install a later version of npm to work around it. It modified the json files and I figured those changes shouldn't be included

**Review by Xavion3 — CHANGES_REQUESTED** [review]
This should be two seperate PRs. Don't include both the engine changes and package changes in the same PR. Also don't include override values which actually affect things in a PR.

**Comment by bennybroseph** [comment]
Already implemented here https://github.com/pagefaultgames/pokerogue/pull/238

**Pull request closed without merging**